external/dynamo/optimized/processor_multilru.py

```
@@ -0,0 +1,833 @@
+                decision_id = data.get("decision_id")
+                break
+
+            # Record routing decision
+            if worker_id is not None:
+                self._metrics.routing_decisions_total.labels(worker_id=str(worker_id)).inc()
+            else:
+                logger.warning("Router stream ended without worker_id; falling back to engine load balancing.")
+
+            return worker_id, decision_id
+
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)